lib/u2/DAOControllerView.es6.js

```diff
@@ -0,0 +1,278 @@
+          return imported;
+        }
+
+        const tableColumns = this.of.getAxiomByName('tableColumns');
+        if (tableColumns.columns) {
+          return tableColumns.columns.map(name => this.of.getAxiomByName(name));
+        }
+
+        return this.of.getAxiomsByClass(foam.core.Property)
+            .filter(prop => !prop.hidden);
+      },
+    },
+    {
+      class: 'FObjectArray',
+      of: 'foam.core.Property',
+      name: 'selected',
+      documentation: `The properties currently selected for rendering in the DAO
+          table view.`,
+      factory: function() {
+        return this.importedSelectedColumns ||
+            Array.from(this.selectable);
+      },
+    },
+    {
+      name: 'columnSelectionE',
+      documentation: `The foam.u2.Element that constitutes the column selection
+          view.`,
+      // TODO(markdittmer): Should this be a ViewSpec instead of a view?
+      factory: function() {
+        return this.OverlayDropdown.create().add(this.columnSelectionView_);
+      },
+    },
+    {
+      name: 'columnSelectionView_',
+      // TODO(markdittmer): Should this be a ViewSpec instead of a view?
+      factory: function() {
+        return this.ToggleListElementsView.create({
+          selectable: this.selectable,
+          selected$: this.selected$,
+        });
+      },
+    },
+    {
+      name: 'columnsIconE',
+      documentation: `The foam.u2.Element that constitutes the column selection
+          action button.`,
+      // TODO(markdittmer): Should this be a ViewSpec instead of a view?
+      factory: function() {
+        return this.E('i').addClass('material-icons').add('more_vert')
+          .addClass(this.myClass('icon')).addClass(this.myClass('button'))
+          .on('click', () => {
+            this.columnSelectionE.open();
+          });
+      },
+    },
+    {
+      class: 'foam.u2.ViewSpec',
+      name: 'searchSpec',
+      documentation: 'The view specification for the search box.',
+      value: {class: 'org.chromium.apis.web.SearchView'},
+    },
+    {
+      name: 'searchE',
+      documentation: 'The foam.u2.Element that constitutes the search box.',
+    },
+    {
+      class: 'Int',
+      name: 'count_',
+    },
+    {
+      class: 'Int',
+      name: 'total_',
+    },
+  ],
+
+  methods: [
+    function init() {
+      this.SUPER();
+      this.data$.dot('data').sub(this.onDAOChange);
+      this.data$.dot('filteredDAO').sub(this.onFilteredDAOChange);
+      this.onDAOChange();
+      this.onFilteredDAOChange();
+    },
+    function initE() {
+      let daoView = this.data.FILTERED_DAO.toE(null, this);
+      daoView.addClass(this.myClass('DAOView'));
+      let searchView = foam.u2.ViewSpec.createView(
+          this.searchSpec, {}, this, this.__subSubContext__)
+          .addClass(this.myClass('search'));
+      this.addClass(this.myClass())
+          .start('div').addClass(this.myClass('controls'))
```

Comment:
There's something to be said about the readability of a 17 line statement. I'd probably start by breaking it up according to the nesting level in the resulting HTML, because trying to match start and end tags in this is more challenging than it should be.

Reply:
Indentation done.